core/App/ehsm_napi.cpp

```diff
@@ -224,41 +254,64 @@ char* NAPI_Decrypt(const char* cmk_base64,
         const char* ciphertext_base64,
         const char* aad)
 {
-    string decode_cmk;
-    string decode_cipher;
-    string plaintext_base64;
-
     RetJsonObj retJsonObj;
     ehsm_status_t ret = EH_OK;
+    ehsm_keyblob_t cmk = {0};
+    ehsm_data_t plaint_data = {0};
+    ehsm_data_t aad_data = {0};
+    ehsm_data_t cipher_data = {0};
+    string plaintext_base64;
+    
+    string cmk_str;
+    string ciphertext_str;
+    int cmk_len;
+    int ciphertext_len;
+    int aad_len;
+    
+    cmk_str = base64_decode(cmk_base64);
+    cmk_len = cmk_str.size();
+    ciphertext_str = base64_decode(ciphertext_base64);
+    ciphertext_len = ciphertext_str.size();
+    aad_len = strlen(aad);
```

Comment:
Have you tested strlen(NULL)?

Reply:
merge @265~275 for simplicity, e.g.
    string cmk_str = base64_decode(cmk_base64);
    int cmk_len = cmk_str.size();
    string ciphertext_str = base64_decode(ciphertext_base64);
    int ciphertext_len = ciphertext_str.size();
    int aad_len = strlen(aad);

Reply:
It is handled in nodejs. If the request parameter does not have aad, an empty string will be passed to napi

Reply:
NULL add is valid.

Reply:
It will be fix.